specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2024-01-01-preview/workspaceRP.json

```diff
@@ -2099,58 +3124,234 @@
+        "name": "EncryptionStatus",
+        "modelAsString": true
+      }
+    },
+    "EncryptionUpdateProperties": {
+      "required": [
+        "keyVaultProperties"
+      ],
+      "type": "object",
+      "properties": {
+        "keyVaultProperties": {
+          "$ref": "#/definitions/EncryptionKeyVaultUpdateProperties"
+        }
+      }
+    },
+    "EndpointDeploymentModel": {
+      "type": "object",
+      "properties": {
+        "format": {
+          "description": "Model format",
+          "type": "string"
+        },
+        "name": {
+          "description": "Model name.",
+          "type": "string"
+        },
+        "source": {
+          "description": "Optional. Deployment model source ARM resource ID.",
+          "type": "string"
+        },
+        "version": {
+          "description": "Model version.",
+          "type": "string"
+        }
+      }
+    },
+    "EndpointDeploymentResourceProperties": {
+      "required": [
+        "type"
+      ],
+      "type": "object",
+      "properties": {
+        "failureReason": {
+          "description": "The failure reason if the creation failed.",
+          "type": "string"
+        },
+        "provisioningState": {
+          "description": "Read-only provision state status property.",
+          "$ref": "#/definitions/DefaultResourceProvisioningState"
+        },
+        "type": {
+          "description": "Kind of the deployment.",
+          "minLength": 1,
+          "pattern": "[a-zA-Z0-9._]",
+          "type": "string"
+        }
+      },
+      "discriminator": "type"
+    },
+    "EndpointDeploymentResourcePropertiesBasicResource": {
+      "required": [
+        "properties"
+      ],
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/Resource"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/EndpointDeploymentResourceProperties"
+        }
+      }
+    },
+    "EndpointDeploymentResourcePropertiesBasicResourceArmPaginatedResult": {
+      "type": "object",
+      "properties": {
+        "nextLink": {
+          "type": "string"
+        },
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/EndpointDeploymentResourcePropertiesBasicResource"
+          }
+        }
+      }
+    },
+    "EndpointKeys": {
+      "type": "object",
+      "properties": {
+        "keys": {
+          "description": "Dictionary of Keys for the endpoint.",
+          "$ref": "#/definitions/AccountApiKeys",
+          "x-ms-secret": true
+        }
+      }
+    },
+    "EndpointModels": {
+      "type": "object",
+      "properties": {
+        "nextLink": {
+          "description": "The link to the next page constructed using the continuationToken.  If null, there are no additional pages.",
+          "type": "string"
+        },
+        "value": {
```

Comment:
Also add suppressions for the following code paths:

{"pluginName":"spectral","extensionName":"@microsoft.azure/openapi-validator","level":"error","message":"Definitions must not have properties named additionalProperties except for user defined tags or predefined references.","code":"AvoidAdditionalProperties","details":{"jsonpath":["definitions","EndpointModels","properties","value","items","properties","capabilities"],"providerNamespace":false,"resourceType":false,"range":{"start":{"line":3276,"column":18},"end":{"line":3277,"column":48}}},"source":[{"document":"[file:///mnt/vss/_work/1/azure-rest-api-specs/specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2024-01-01-preview/workspaceRP.json](file:///mnt/vss/_work/1/azure-rest-api-specs/specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2024-01-01-preview/workspaceRP.json)","position":{"line":3276,"column":11}}]}

{"pluginName":"spectral","extensionName":"@microsoft.azure/openapi-validator","level":"error","message":"Definitions must not have properties named additionalProperties except for user defined tags or predefined references.","code":"AvoidAdditionalProperties","details":{"jsonpath":["definitions","EndpointModels","properties","value","items","properties","finetuneCapabilities"],"providerNamespace":false,"resourceType":false,"range":{"start":{"line":3276,"column":18},"end":{"line":3277,"column":48}}},"source":[{"document":"[file:///mnt/vss/_work/1/azure-rest-api-specs/specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2024-01-01-preview/workspaceRP.json](file:///mnt/vss/_work/1/azure-rest-api-specs/specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2024-01-01-preview/workspaceRP.json)","position":{"line":3276,"column":11}}]}

Reply:
- $.definitions.EndpointModels.properties.value.items.$ref
      
      I have the above patch added already, I will make the scope larger to $.definitions.EndpointModels.properties.value.items and see if it will work

Reply:
@visingla-ms 

I guess no luck, even if I make the scope larger the LintError pipeline is still complaining about it.

Reply:
Why not update these two as shown below:

$.definitions.EndpointModels.properties.value.items.properties.capabilities
$.definitions.EndpointModels.properties.value.items.properties.finetuneCapabilities

?

Reply:
Oh that's the format, I thought the $ref should point to the exact line of 3276, let me update the PR and try, thanks for the help

Reply:
@visingla-ms Thanks a lot for the help it's working!